src/providers/ContentShareProvider/state.tsx

```diff
@@ -11,71 +13,177 @@ export enum ContentActionType {
   REMOVE,
   DENIED,
   RESET,
+  UPDATE_CAN_START,
 }
 
-type StartingAction = {
-  type: ContentActionType.STARTING;
-  payload?: any;
-};
-
-type DidStopAction = {
-  type: ContentActionType.DID_STOP;
-  payload?: any;
-};
+type EmptyPayload = Record<string, never>;
 
-type UpdateAction = {
-  type: ContentActionType.UPDATE;
-  payload: UpdatePayload;
+type StartingPayload = EmptyPayload;
+type DidStopPayload = {
+  localAttendeeId: string;
 };
-
 type UpdatePayload = {
   isLocalUser: boolean;
   tileState: VideoTileState;
 };
-
-type TogglePauseAction = {
-  type: ContentActionType.TOGGLE_PAUSE;
-  payload?: any;
-};
-
-type RemoveAction = {
-  type: ContentActionType.REMOVE;
-  payload: number;
-};
-
-type DeniedAction = {
-  type: ContentActionType.DENIED;
-  payload?: any;
+type TogglePausePayload = EmptyPayload;
+type RemovePayload = {
+  tileId: number;
+  isLocalUser: boolean;
 };
-
-type ResetAction = {
-  type: ContentActionType.RESET;
-  payload?: any;
+type DeniedPayload = EmptyPayload;
+type ResetPayload = EmptyPayload;
+type UpdateCanStartPayload = {
+  maxContentShares: number;
 };
 
+// Action types
 export type Action =
-  | StartingAction
-  | DidStopAction
-  | UpdateAction
-  | TogglePauseAction
-  | RemoveAction
-  | DeniedAction
-  | ResetAction;
+  | { type: ContentActionType.STARTING; payload?: StartingPayload }
```

Comment:
We are updating some of the types for the payload here. What if someone is depending on this component currently and it starts failing for them as the types updated. Is this a backward incompatible change?

Reply:
These are internal type, so it's fine. It's `export` here but not `export` in index.ts file. It's not public available.